src/ui/common/src/utils/workflows.tsx

```diff
@@ -26,6 +26,63 @@ export type RetentionPolicy = {
   k_latest_runs: number;
 };
 
+export type WorkflowMetrics = {
+  id: string;
+  description: string;
+  from: string;
+  name: string;
+  to: string;
+  result: {
+    content_path: string;
+    // This is the thing that we want to show in the table view.
+    content_serialized: string;
+    exec_state: ExecState;
+    serialization_type: string;
+  };
+};
+/*
+{
+    error?: string;
+    failure_type?: string;
+    status: string;
+    timestamps: {
+      finished_at?: string;
+      pending_at?: string;
+      resgistered_at?: string;
+      running_at?: string;
+      user_logs: {
+        stderr: string;
+        stdout: string;
+      }
+    }
+  }*/
+export type ExecutionResult = {
+  id: string;
+  exec_state: ExecState;
+};
+
+export type WorkflowChecks = {
```

Comment:
the intention was to reuse `common/src/handlers/responses/artifact/ArtifactResultResponse`. It seems that the `spec` field is missing, so may need a little work adding that field in `ArtifactResultResponse` object.